scripts/Tag-TeamCityBuild.ps1

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env pwsh
+
+<#
+.SYNOPSIS
+    Tags the given TeamCity build with the given tag.
+.PARAMETER BuildId
+    The BuildId of the build to be tagged.
+.PARAMETER Tag
+    The tag to put on this build.
+#>
+
+[cmdletbinding(SupportsShouldProcess = $true)]
+param(
+    [Parameter(Mandatory = $true)]
+    [string]$BuildId,
+    [Parameter(Mandatory = $true)]
+    [string]$Tag,
+    [Parameter(Mandatory = $true)]
+    [string]$UserName,
+    [Parameter(Mandatory = $true)]
+    [string]$Password
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 2
+
+$authInfo = "${UserName}:$Password"
+$authEncoded = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes($authInfo))
+$basicAuthValue = "Basic $authEncoded"
+
+$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
```

Comment:
Use a powershell map:
```ps1
$headers = @{
    Authorization  = "Basic $authEncoded"
	"Content-Type" = "text/plain"
}
```

Reply:
Huh, it works for me, but I'm using PowerShell 6. What are you using?

The quotes are necessary because dash is an invalid character in PS identifier syntax, so we need to make the content-type entry a string key instead.

Reply:
I was imprecise in my wording, I meant that when I did `Content-Type` without the quotes it failed. It didn't give me a very descriptive error message so I didn't know the `-` was the problem. Thanks!